communitiesuk/funding-service-design-docker-runner
# Automate fund-round creation

Automates the whole setting up Apply and Assess to use a FAB created fund, round and form. 
Lots of our code is moved at the moment, so this will soon break, but if there is any value in this script, we will update it.

Pre-requisites:
1. Be checked out on `main` on the following repos, pull latest changes, stash uncommitted changes. The script will create a new branch on multiple repos with the required config.
2. You can either create your own FAB fund/round/form or ask for a sample; you need to extract the config from FAB and paste it in `/fund-round/.`. 
3. You might need to clean up a bit FAB's exported stuff. Name the dir nicely without special chars/spaces.

Optional step:
Do a `make down`. Keep things clean.

### How to test
1. Make sure your FAB directory with the `html`, `fund-store` and `form-runner` config are inside `/fund-round`
2. Run `./scripts/fund-round-creator.sh` and follow the prompt
3. Until [PR](https://github.com/communitiesuk/funding-service-pre-award-stores/pull/13) is merged you will have to run seed the fund manually: 
```bash
docker exec funding-service-pre-award-stores-1 python -m fund_store.scripts.fund_round_loaders.load_fund_round_from_fab --fund_short_code <fund-short-name-uppercase>
```
4. `make up`

You can now go to:
https://frontend.levellingup.gov.localhost:3008/funding-round/fund-short-name/round-short-name
Create a quick application.

You can now go to:
https://assessment.levellingup.gov.localhost:3010
You can see you application there.
